File: Floyd/floyd.py
def Floyd(v, D):
    for k in range(0, v):
        for i in range(0, v):
            for j in range(0, v):
                if (D[i][k] != 0) and (D[k][j] != 0) and (i != j):
                    if (D[i][k]+D[k][j] < D[i][j]) or (D[i][j] == 0):
                        D[i][j] = D[i][k] + D[k][j]
    return v, D


def ReadData():
    f = open('input.txt', 'r')
    l = [line.strip() for line in f]
    v = int(l[0])
    D=[]
    for i in range(1, v+1):
        print (l[i])
        sD=[]
        v1, v2, v3 = map(int, l[i].split())
        sD.append(v1)
        sD.append(v2)
        sD.append(v3)
        D.append(sD)       
    f.close
    return v,  D
    

def WriteData(v, D):
    f = open('output.txt', 'w')
    for i in range(0, v):
        for j in range(0, v):
            f.write(str(D[i][j])+" ")
        f.write('\n')
    f.close


# main block
v, D = ReadData()
v, D = Floyd(v, D)
WriteData(v, D)

File: Floyd/floyd.cpp
#include <fstream>
#include <iostream>

const int16_t MAXN = 100; 

int64_t D[MAXN][MAXN];
int64_t V; 

//алгоритм Флойда-Воршелла
void FU()
{
	for (int64_t i=0; i<V; i++) D[i][i]=0; 
	for (int64_t k=0; k<V; k++)
		for (int64_t i=0; i<V; i++)
			for (int64_t j=0; j<V; j++)
				if (D[i][k] && D[k][j] && i!=j)
					if (D[i][k]+D[k][j]<D[i][j] || D[i][j]==0)
						D[i][j]=D[i][k]+D[k][j]; 
}

void ReadData(){
    using namespace std;
    int64_t m, v1, v2, v3;
    ifstream fin("input.txt");
    fin >> V;
    for (int64_t i=1; i<=V; i++){
        fin >> v1 >> v2 >> v3;
        D[i-1][0] = v1;
        D[i-1][1] = v2;
        D[i-1][2] = v3;
    }
    fin.close();
}

void WriteData(){
    std::ofstream fout("output.txt");
    for (int64_t i=0; i<V; i++){
		for (int64_t j=0; j<V; j++){
            fout<<D[i][j]<<" ";
        }
        fout<<"\n";
    }
    fout.close();
}

//Головна функція
int main(){
    ReadData();
    FU();
    WriteData();
    return 0;
}
